### PR TITLE
Support building on NetBSD.

### PR DESCRIPTION
**Pull request by niacat**
#378 can be closed.

### CODE REVIEW

**Comment by AJenbo** [comment]
Oh forgot about NetBSD, wonder if we are missing any other BSDs?

Cool that you have UBSAN

File: README.md [hunk 1]
```diff
@@ -63,6 +63,19 @@ cmake ..
...
 </details>
+<details><summary>NetBSD</summary>
+
+### Installing dependencies
+```
+pkgin install cmake SDL2_mixer SDL2_ttf libsodium
+```
+### Compiling
+```
+cd build
```
Comment (glebm) [inline README.md:74]: If cmake is reasonably recent on NetBSD, `cd` can be avoided:

```bash
cmake -S. -Bbuild ..
cmake --build build -j $(sysctl -n hw.ncpu)
```

Comment (niacat) [reply README.md:74]: Cool. It should be the most recent cmake release.

**Comment by niacat** [comment]
@AJenbo We've also got ASAN but building with that I got a segfault immediately after launching devilutionx, unfortunately

Every currently maintained BSD descendant should now be covered by the CMakeLists, there's some older ones that aren't maintained any more though.

**Comment by kraileth** [comment]
Well, MidnightBSD is actually actively maintained and has it's own kernel name, but it's extremely exotic. The others that I can think of are either customized versions of one of the big others (and don't use a different kernel ident) or are obviously / most likely inactive.

**Pull request merged**